synergy-design-system/synergy-design-system
# feat: ✨ create syn-file

# Pull Request

## 📖 Description

This PR adds the new `<syn-file>` component that may act as a replacement for `<input type="file">`, as well as a custom dropzone.

The following issues must still be resolved before this can enter testing:

- [x] There is currently no template in Storybook. We will have to work out where we put it. (Will be tackled in #569)
- [x] When using `droparea`, you are always able to upload multiple files. This should be fixed if possible.
- [x] Tests are missing nearly completely
- [x] Must be added to framework demos in a meaningful way.
- [x] Design alignment is not 100% through.
- [x] The component implements `SynergyFormsModule`, but constraint validation is currently an empty shell.
- [x] Expose all events needed (e.g. drag&drop, file)
- [x] There is currently a custom defined `hover` state for the dropzone that has to be aligned with design.
- [x] There is currently no disabled state while drag&drop is in place.
- [x] Storybook Figma links are currently broken and pointing to the wrong story.
- [x] 🔥 Completely compatible with native `<input type="file">` by leveraging `formData` and binary data values

### 🎫 Issues

Closes #402 

## 👩‍💻 Reviewer Notes

- There where a11y errors for the placeholder (contrast ratio of 2.4 instead of 4.5). Has to be talked about with design. I disabled the rule for the time being.
- As there is no possibility to programmatically set the value of a inputs `files` object, there is no way for two way databindings for angular or vue here. This is a limitation of the platform, not of the component itself.

## 📑 Test Plan

- Have a look at the provided stories
- Have a look at the framework demos to see how it can be used in the "real" world.
- Make also sure to test with keyboard navigation.
- Check if the static translations make sense

## ✅ DoD
